src/backsolve_adjoint.jl

```diff
@@ -205,16 +216,25 @@ end
 
 @noinline function SDEAdjointProblem(sol, sensealg::BacksolveAdjoint,
                                      t = nothing,
-                                     dg_discrete::DG1 = nothing,
-                                     dg_continuous::DG2 = nothing,
+                                     dgdu_discrete::DG1 = nothing,
+                                     dgdp_discrete::DG2 = nothing,
+                                     dgdu_continuous::DG3 = nothing,
+                                     dgdp_continuous::DG4 = nothing,
                                      g::G = nothing;
                                      checkpoints = sol.t,
                                      callback = CallbackSet(),
                                      corfunc_analytical = nothing, diffusion_jac = nothing,
                                      diffusion_paramjac = nothing,
-                                     kwargs...) where {DG1, DG2, G}
-    dg_discrete === nothing && dg_continuous === nothing && g === nothing &&
-        error("Either `dg_discrete`, `dg_continuous`, or `g` must be specified.")
+                                     kwargs...) where {DG1, DG2, DG3, DG4, G}
+    dgdu_discrete === nothing && dgdu_continuous === nothing && g === nothing &&
+        error("Either `dgdu_discrete`, `dgdu_continuous`, or `g` must be specified.")
+    t !== nothing && dgdu_discrete === nothing && dgdp_discrete === nothing &&
+        error("It looks like you're using the direct `adjoint_sensitivities` interface
+               with a discrete cost function but no specified `dgdu_discrete` or `dgdp_discrete`.
+               Please use the higher level `solve` interface or specify these two contributions.")
+
+    dg_discrete = (dgdu_discrete, dgdp_discrete)
+    dg_continuous = (dgdu_continuous, dgdp_continuous)
```

Comment:
why put them input tuples instead of keeping them separate?

Reply:
I thought it reads slightly easier internally if we have, e.g. only `pg`, `pg_config` and `dg_val` in: 
```julia
pgpu = UGradientWrapper(g, tspan[2], p)
pgpu_config = build_grad_config(sensealg, pgpu, u0, p)
pgpp = ParamGradientWrapper(g, tspan[2], u0)
pgpp_config = build_grad_config(sensealg, pgpp, p, p)
pg = (pgpu, pgpp)
pg_config = (pgpu_config, pgpp_config)
dg_val = (similar(u0, numindvar), similar(u0, numparams))
dg_val[1] .= false
dg_val[2] .= false
```
in the `diffcache` and was not entirely sure where to stop (discrete and continuous never mix in a function, that's why I thought keeping them grouped from the beginning might be a good idea .. but yeah in the end we need every time the unpacked version of `dg_discrete` and `dg_continuous`, respectively.)